src/main/java/teammates/ui/controller/InstructorSearchPageData.java

```diff
@@ -58,4 +139,272 @@ public String getStudentRecordsLink(String courseId, StudentAttributes student){
         link = addUserIdToUrl(link);
         return link;
     }
+    
+    private void setSearchCommentsForStudentsTables() {
+        searchCommentsForStudentsTables = new ArrayList<SearchCommentsForStudentsTable>();       
+        for (String giverEmailPlusCourseId : commentSearchResultBundle.giverCommentTable.keySet()) {
+            searchCommentsForStudentsTables.add(createSearchCommentsForStudentsTable(giverEmailPlusCourseId));
+        }
+    }
+    
+    private void setSearchCommentsForResponsesTables() {
+        searchCommentsForResponsesTables = new ArrayList<SearchCommentsForResponsesTable>();
+        searchCommentsForResponsesTables.add(createSearchCommentsForResponsesTable());
+    }
+    
+    private void setSearchStudentsTables() {
+        searchStudentsTables = new ArrayList<SearchStudentsTable>(); // 1 table for each course      
+        ArrayList<String> courseIdList = getCourseIdsFromStudentSearchResultBundle(studentSearchResultBundle.studentList);
+        
+        for (String courseId : courseIdList) {
+            searchStudentsTables.add(createSearchStudentsTable(courseId));
+        }
+    }
+    
+    private ArrayList<String> getCourseIdsFromStudentSearchResultBundle(
+                                    List<StudentAttributes> studentList) {
+        ArrayList<String> courses = new ArrayList<String>();
```

Comment:
In other places it is List<SomeClass> list = new ArrayList<SomeClass>()
Can change to List in line 166